### PR TITLE
Add test, comment, cleanup for DNSPrefix webhook fix

### PR DESCRIPTION
**Pull request by nojnhuh**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

This PR follows up from #4308 to address outstanding comments.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

- [X] cherry-pick candidate

**TODOs**:

- [X] squashed commits
- [ ] includes documentation
- [X] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm

File: api/v1beta1/azuremanagedcontrolplane_webhook_test.go [hunk 1]
```diff
@@ -2559,6 +2559,29 @@ func TestAzureManagedControlPlane_ValidateUpdate(t *testing.T) {
 			},
 			wantErr: true,
 		},
+		{
+			name: "AzureManagedControlPlane DNSPrefix is immutable ok nil with cluster name",
```
Comment (mboersma) [inline api/v1beta1/azuremanagedcontrolplane_webhook_test.go:2563]:
```suggestion
			name: "AzureManagedControlPlane DNSPrefix is immutable or nil with cluster name",
```

Comment (nojnhuh) [reply api/v1beta1/azuremanagedcontrolplane_webhook_test.go:2563]: I agree this name is bad as-is but I might tweak it slightly from your suggestion.

**Review by willie-yao — APPROVED** [review]
/lgtm

**Review by mboersma — APPROVED** [review]
/lgtm
/approve

**Pull request merged**